mbed_host_tests/host_tests/default_auto.py

```diff
@@ -15,32 +15,32 @@
 limitations under the License.
 """
 
-from sys import stdout
+
 from mbed_host_tests import BaseHostTest
 
+
 class DefaultAuto(BaseHostTest):
     """ Simple, basic host test's test runner waiting for serial port
         output from MUT, no supervision over test running in MUT is executed.
     """
-    def test(self, selftest):
-        result = selftest.RESULT_SUCCESS
-        try:
-            line = ""
-            while True:
-                c = selftest.mbed.serial_read(512)
-                if c is None:
-                    return selftest.RESULT_IO_SERIAL
-                stdout.write(c)
-                stdout.flush()
-
-                line += c
-                if '\n' in line:
-                    if '{end}' in line:
-                        return None # Run executable end
-                    i = line.index('\n')
-                    line = line[i:]
-
-        except KeyboardInterrupt, _:
-            selftest.notify("\r\n[CTRL+C] exit")
-            result = selftest.RESULT_ERROR
-        return result
+
+    result = None
+
+    def _callback_end(self, key, value, timeout):
+        if value == 'success':
+            self.result = True
+        elif value == 'failure':
+            self.result = False
+
+    def _callback_exit(self, key, value, timeout):
+        self.notify_complete()
+
+    def setup(self):
+        self.register_callback('end', self._callback_end)
+        self.register_callback('exit', self._callback_exit)
+
+    def test(self):
+        return self.result
+
+    def teardown(self):
```

Comment:
I would say default auto is with only test() implemented, setup/teardown is not default, is it?

Or is it still as you described yesterday, test() is the method called after HostTest receives ```notify_complete()``` ?

I would imagine test() to be

```
def test(self):
   # setting timeout for this test case , it's not the same as host test timeout set by a client which is setting timeout for entire test unit. Or?
   self.set_timeout(5);
   self.register_callback(.....)
   # here I return, a timeout is set, means this test() is asynch, wait for notify_complete() or timeout. 
   # notify_complete can have an argument which would indicate if it is Pass or Fail
```